homeassistant/components/downloader/__init__.py

```diff
@@ -57,122 +86,143 @@ def setup(hass: HomeAssistant, config: ConfigType) -> bool:
         _LOGGER.error(
             "Download path %s does not exist. File Downloader not active", download_path
         )
-
         return False
 
-    def download_file(service: ServiceCall) -> None:
-        """Start thread to download file specified in the URL."""
-
-        def do_download() -> None:
-            """Download the file."""
-            try:
-                url = service.data[ATTR_URL]
-
-                subdir = service.data.get(ATTR_SUBDIR)
+    lock = asyncio.Lock()
 
-                filename = service.data.get(ATTR_FILENAME)
+    async def _async_run_download(call: ServiceCall) -> None:
+        async with lock:
```

Comment:
Why is the mutex added? It has nothing to do with adding a config flow, please do it in a separate PR instead.